packages/frontend-test-utils/src/app/renderInTestApp.tsx

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  coreExtensionData,
+  createExtension,
+} from '@backstage/frontend-plugin-api';
+import { createExtensionTester } from './createExtensionTester';
+
+/**
+ * @public
+ * Renders the given element in a test app, for use in unit tests.
+ */
+export function renderInTestApp(element: JSX.Element) {
```

Comment:
I would appreciate recommendations on what options would be good to receive here 🙂 .

Reply:
Basically all of our existing test code uses `renderInTestApp(element)`. Searching for `renderInTestApp\([A-Z]` I only find a single test outside of the tests for `renderInTestApp` itself: https://github.com/backstage/backstage/blob/0296f272b40d18bd1df8c0922737df8eb8473187/plugins/cost-insights/src/components/ProjectSelect/ProjectSelect.test.tsx#L45-L50

That's a kinda weird pattern too, and is ofc trivial to refactor to use an element instead and even makes the test a bit cleaner.

It's also the case that `render` from RTL only receives and `ReactElement` as well, so I think we should stick to that.

The design of our current `renderInTestApp` came about because we had this whole builder that allowed you to add various different app contexts gradually and injects HOC-style props like themes and whatnot. Over time that all just got simplified down to React contexts and always rendering in the entire app, so I don't think the component approach has much value anymore.

In the end the intention of `renderInTestApp` is to be a simple utility that renders a React element in the context of a Backstage app, where it has access to all the core Utility APIs, theme and routing contexts and whatever else one might expect.